### PR TITLE
Allow external clients (e.g UI) to discover supported authentication methods

### PR DESCRIPTION
**Pull request by shaan1337**
Added: Allow external clients to discover supported authentication methods

Related to: https://github.com/EventStore/EventStore.UI/issues/265
This PR adds an `authentication` property to the JSON response of `/info` to allow external clients (e.g UI) to discover supported authentication methods (insecure, internal, oauth, ldaps)

The PR also adds the `WWW-Authenticate` header to 401 responses but with an `X-` in front of the authentication scheme to prevent default browser behaviour (like basic auth popups)

Note: It requires EventStore.Plugins 20.6.1-preview3 to be published (https://github.com/EventStore/EventStore.Plugins/pull/10)

### CODE REVIEW

File: src/EventStore.Core/ClusterVNode.cs [hunk 1]
```diff
@@ -472,26 +472,38 @@ public class ClusterVNode :
...
+						httpAuthenticationProviders.Add(new BasicHttpAuthenticationProvider(_authenticationProvider));
+						break;
+					case "Bearer":
+						httpAuthenticationProviders.Add(new BearerHttpAuthenticationProvider(_authenticationProvider));
+						break;
+					case "Insecure":
+						httpAuthenticationProviders.Add(new PassthroughHttpAuthenticationProvider(_authenticationProvider));
+						break;
+					default:
+						Log.Warning($"Unsupported Authentication Scheme: {authenticationScheme}");
```
Comment (hayley-jean) [inline src/EventStore.Core/ClusterVNode.cs:490]: How would Event Store react if none of the authentication schemes were supported? Is that something we want to check after this loop?

Comment (shaan1337) [reply src/EventStore.Core/ClusterVNode.cs:490]: @hayley-jean we could log an error in the case where none of the auth schemes are supported, what do you think?

cc @jageall

Comment (hayley-jean) [reply src/EventStore.Core/ClusterVNode.cs:490]: Would this be a case where we would want to log an error and shut the node down?

Comment (jageall) [reply src/EventStore.Core/ClusterVNode.cs:490]: No supported auth scheme should result in the server shutting down

Comment (shaan1337) [reply src/EventStore.Core/ClusterVNode.cs:490]: @hayley-jean @jageall Fixed it in this commit: 0ebef5cf9c1515e918ac8151db8bb931b30ce777, thanks !

**Review by hayley-jean — APPROVED** [review]

**Review by hayley-jean — APPROVED** [review]

**Review by jageall — APPROVED** [review]

**Pull request merged**